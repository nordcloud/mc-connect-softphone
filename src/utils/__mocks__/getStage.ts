export const getStage = jest.fn().mockReturnValue('prod');
