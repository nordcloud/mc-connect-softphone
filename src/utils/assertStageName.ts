import { stageConsts } from '../consts';
import { Stage } from '../types';

export function assertStageName(stage?: string): asserts stage is Stage {
  if (stage === undefined) {
    throw new Error('Missing STAGE');
  }

  if (!Object.keys(stageConsts).includes(stage as Stage)) {
    throw new Error('Invalid STAGE stage');
  }
}
