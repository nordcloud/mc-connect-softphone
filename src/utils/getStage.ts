import { Stage } from '../types';
import { assertStageName } from './assertStageName';

export function getStage(): Stage {
  const { STAGE } = process.env;

  assertStageName(STAGE);

  return STAGE;
}
