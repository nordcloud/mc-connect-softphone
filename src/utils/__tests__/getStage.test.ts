import { assertStageName } from '../assertStageName';
import { getStage } from '../getStage';

jest.mock('../assertStageName');

process.env.STAGE = 'dummyStageName';

describe('getStage', () => {
  it('calls assertStageName with correct args', () => {
    getStage();
    expect(jest.mocked(assertStageName).mock.calls).toMatchSnapshot();
  });

  it('returns a correct value', () => {
    expect(getStage()).toMatchSnapshot();
  });
});
