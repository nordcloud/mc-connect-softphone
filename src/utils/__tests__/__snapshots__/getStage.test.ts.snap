// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStage calls assertStageName with correct args 1`] = `
[
  [
    "dummyStageName",
  ],
]
`;

exports[`getStage returns a correct value 1`] = `"dummyStageName"`;
